conan/tools/cmake/toolchain.py

```diff
@@ -654,7 +654,9 @@ def remove(self, name):
         del self._blocks[name]
 
     def __setitem__(self, name, block_type):
-        self._blocks[name] = block_type(self._conanfile, self._toolchain)
+        # Create a new class inheriting Block with the elements of the provided one
+        new_class = type('proxyUserBlock', (Block,), dict(block_type.__dict__))
```

Comment:
This should be enough, right?

Reply:
Yes, I think so, just need to fix the Block test that is doing ``super()``

Reply:
Please, @memsharded  check the last commit. Is it worth it or should we drop the possibility of inheritance from blocks completely?

Reply:
Good point, I'd say keep it simple at the moment, lets remove the inheritance case, sounds that with .values[] it is easier to achieve the same, and if users request it, we will re-consider.